library/core/src/convert/num.rs

```diff
@@ -109,7 +111,7 @@ impl_from!(u8 => u16, #[stable(feature = "lossless_int_conv", since = "1.5.0")])
 impl_from!(u8 => u32, #[stable(feature = "lossless_int_conv", since = "1.5.0")]);
 impl_from!(u8 => u64, #[stable(feature = "lossless_int_conv", since = "1.5.0")]);
 impl_from!(u8 => u128, #[stable(feature = "i128", since = "1.26.0")]);
-impl_from!(u8 => usize, #[stable(feature = "lossless_int_conv", since = "1.5.0")]);
+impl_from!(u8 => usize, #[stable(feature = "lossless_int_conv", since = "1.5.0")], #[cfg_attr(flux, flux::spec(fn(x:u8) -> usize[x]))]);
```

Comment:
Can you outline a path on how this might eventually get integrated upstream?

Reply:
Could you clarify whether you're referring to specs in general or just the spec in this macro?

Reply:
Actually just this one: to me (and I could be wrong!) it looks like a change that, if we were to try to do more proofs using Flux, would require changes in lots of places. For other specs/contracts there is a path in that Rust upstream has experimental contracts support. But tool-specific annotations seem a lot less likely to be merged upstream.

Reply:
You are not wrong in that more proofs will require more annotations. We've thought about this, and one option is to provide signatures in a separate file (à la ocaml .mli files), but this is something we have not implemented yet (we have something similar for [extern specs](https://flux-rs.github.io/flux/guide/specifications.html#extern-specs)).

Reply:
Sadly, no, we try that first, but the macro is also used to convert from bool to integers, and we can't use the same signature for that, i.e., the following is _ill-formed_ when `$Small` is `bool`.

```rust
#[cfg_attr(flux, flux::spec(fn(x: $Small) -> $Large[x])
```

Reply:
Yes, the catch here is that we want to branch on the `$Small` and `$Large` and only put _different_ specs depending on their values. 

There is perhaps some way to do this with the macros but we couldn't quite work it out, unfortunately.

Would be great if you have any suggestions.

Reply:
Hi @tautschnig -- we have an update to `flux` which would simplify this PR so that it just looks [like this](https://github.com/model-checking/verify-rust-std/compare/main...flux-rs:verify-rust-std:flux-submission-with-cast#diff-154dff23a2c2f4fb97915229fd9c27d73293940250088020bfd2074094186ce4) 

That is, as you suggest, which moves the flux spec

> into the macro_rules so that all generated implementations feature a flux_spec?

Would you prefer us to update this PR with this change?

Reply:
@ranjitjhala Apologies for not getting back to you on this one. I do like the new proposal better as it is constrained to a much more local change. Would you mind submitting this change as a follow-up PR?

Reply:
will do, thanks!